File: app/src/main/java/pl/softfly/flashcards/entity/app/AppConfig.java
package pl.softfly.flashcards.entity.app;

import androidx.annotation.NonNull;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

import java.util.Arrays;
import java.util.List;

/**
 * @author Grzegorz Ziemski
 */
@Entity
public class AppConfig {

    public static final String OFF = "Off";

    public static final String DARK_MODE = "DarkMode";

    public static final String DARK_MODE_DEFAULT = "System";

    public static final String DARK_MODE_ON = "On";

    public static final String DARK_MODE_OFF = "Off";

    public static final List<String> DARK_MODE_OPTIONS = Arrays.asList(DARK_MODE_DEFAULT, DARK_MODE_ON, DARK_MODE_OFF);

    public static final String EDGE_BAR_Off = "Off";

    public static final String EDGE_BAR_SHOW_LEARNING_STATUS = "LearningStatus";

    public static final String EDGE_BAR_SHOW_RECENTLY_SYNCED = "RecentlySynced";

    public static final String LEFT_EDGE_BAR = "LeftEdgeBar";

    public static final String LEFT_EDGE_BAR_DEFAULT = EDGE_BAR_SHOW_LEARNING_STATUS;

    public static final String RIGHT_EDGE_BAR = "RightEdgeBar";

    public static final String RIGHT_EDGE_BAR_DEFAULT = EDGE_BAR_SHOW_RECENTLY_SYNCED;


    @PrimaryKey
    @NonNull
    private String key;

    private String value;

    public AppConfig(String key, String value) {
        this.key = key;
        this.value = value;
    }

    public String getKey() {
        return key;
    }

    public void setKey(String key) {
        this.key = key;
    }

    public String getValue() {
        return value;
    }

    public void setValue(String value) {
        this.value = value;
    }
}

File: room/src/main/java/pl/gocards/room/entity/app/AppConfig.kt
package pl.gocards.room.entity.app

import androidx.room.Entity
import androidx.room.PrimaryKey

/**
 * @author Grzegorz Ziemski
 */
@Entity
data class AppConfig(
    @PrimaryKey
    var key: String,
    var value: String
) {
    companion object {
        const val DARK_MODE = "DarkMode"
        const val DARK_MODE_DEFAULT = "System"
        const val DARK_MODE_ON = "On"
        const val DARK_MODE_OFF = "Off"
        val DARK_MODE_OPTIONS = listOf(DARK_MODE_DEFAULT, DARK_MODE_ON, DARK_MODE_OFF)
        const val EDGE_BAR_OFF = "Off"
        const val EDGE_BAR_SHOW_LEARNING_STATUS = "LearningStatus"
        const val EDGE_BAR_SHOW_RECENTLY_SYNCED = "RecentlySynced"
        const val LEFT_EDGE_BAR = "LeftEdgeBar"
        const val LEFT_EDGE_BAR_DEFAULT = EDGE_BAR_SHOW_LEARNING_STATUS
        const val RIGHT_EDGE_BAR = "RightEdgeBar"
        const val RIGHT_EDGE_BAR_DEFAULT = EDGE_BAR_SHOW_RECENTLY_SYNCED
        const val PREMIUM = "Premium"
        const val FIRST_USED_AT = "FirstUsedAt"
        const val LAST_EXCEPTION_AT = "LastExceptionAt"
        const val EXPLORE_POLL_COMPLETED = "ExplorePollCompleted"
    }
}